### PR TITLE
Strip html tags

### PR DESCRIPTION
**Pull request by jinxiwu**
parse the string into a dom object and go through every node to get the InnerText

### CODE REVIEW

File: CSharp/Samples/Microsoft.Bot.Builder.Abstractions.Teams/TeamsActivityProcessor.cs [hunk 1]
```diff
@@ -260,5 +266,37 @@ private async Task<InvokeResponse> ProcessTeamsInvokeActivityAsync(ITurnContext
 
             return await this.invokeActivityHandler.HandleInvokeTaskAsync(turnContext).ConfigureAwait(false);
         }
+
+        private static string stripHtmlTag(string content)
+        {
+            var doc = new HtmlDocument();
+            doc.LoadHtml(content);
+            var TextRestrictedHtmlTags = new HashSet<string> { "at", "b", "i", "ss", "font", "u", "s", "pre" };
```
Comment (robin-liao) [inline CSharp/Samples/Microsoft.Bot.Builder.Abstractions.Teams/TeamsActivityProcessor.cs]: I believe only `<at>` and `<attachment>` are enough. What's the other "hooks" being used in "body" that are not part of regular html? @RamjotSingh

Comment (robin-liao) [reply CSharp/Samples/Microsoft.Bot.Builder.Abstractions.Teams/TeamsActivityProcessor.cs]: Then let's only keep the 2 elements for now, while stripe the others. (

File: CSharp/Samples/Microsoft.Bot.Builder.Abstractions.Teams/TeamsActivityProcessor.cs [hunk 2]
```diff
@@ -260,5 +266,37 @@ private async Task<InvokeResponse> ProcessTeamsInvokeActivityAsync(ITurnContext
...
+            string result = "";
+            if (tags.Contains(node.Name))
+            {
+                result += node.OuterHtml;
+            }
+            else
+            {
+                foreach (HtmlNode childNode in node.ChildNodes)
+                {
+                    if (childNode.NodeType == HtmlNodeType.Text)
```
[the remark this answers is not in the document]
Comment (jinxiwu) [reply CSharp/Samples/Microsoft.Bot.Builder.Abstractions.Teams/TeamsActivityProcessor.cs]: checked and their HtmlNodeType are not Text

File: CSharp/Samples/Microsoft.Bot.Builder.Abstractions.Teams/TeamsActivityProcessor.cs (hunk 1 above)
[the remark this answers is not in the document]
Comment (jinxiwu) [reply CSharp/Samples/Microsoft.Bot.Builder.Abstractions.Teams/TeamsActivityProcessor.cs]: done

Comment (jinxiwu) [reply CSharp/Samples/Microsoft.Bot.Builder.Abstractions.Teams/TeamsActivityProcessor.cs]: fixed

File: CSharp/Samples/Microsoft.Bot.Builder.Abstractions.Teams/TeamsActivityProcessor.cs [hunk 3]
```diff
@@ -260,5 +266,37 @@ private async Task<InvokeResponse> ProcessTeamsInvokeActivityAsync(ITurnContext
 
             return await this.invokeActivityHandler.HandleInvokeTaskAsync(turnContext).ConfigureAwait(false);
         }
+
+        private static string stripHtmlTag(string content)
```
[the remark this answers is not in the document]
Comment (jinxiwu) [reply CSharp/Samples/Microsoft.Bot.Builder.Abstractions.Teams/TeamsActivityProcessor.cs]: fixed

[the remark this answers is not in the document]
Comment (jinxiwu) [reply CSharp/Samples/Microsoft.Bot.Builder.Abstractions.Teams/TeamsActivityProcessor.cs]: fixed

File: CSharp/Samples/Microsoft.Bot.Builder.Abstractions.Teams/TeamsActivityProcessor.cs [hunk 4]
```diff
@@ -239,13 +246,17 @@ private async Task<InvokeResponse> ProcessTeamsInvokeActivityAsync(ITurnContext
             }
 
             if (teamsContext.IsRequestMessagingExtensionFetchTask())
-            {
-                return await this.invokeActivityHandler.HandleMessagingExtensionFetchTaskAsync(turnContext, teamsContext.GetMessagingExtensionActionData()).ConfigureAwait(false);
+            {   
+                MessagingExtensionAction messagingExtensionActionData = teamsContext.GetMessagingExtensionActionData();
```
Comment (RamjotSingh) [inline CSharp/Samples/Microsoft.Bot.Builder.Abstractions.Teams/TeamsActivityProcessor.cs]: Do not modify the content of the incoming request. User can choose to strip the html or not. Modifying the model is not right at all.

File: CSharp/Microsoft.Bot.Schema.Teams/Generated/MessageActionsPayloadBody.cs [hunk 5]
```diff
@@ -57,5 +60,12 @@ public MessageActionsPayloadBody(string contentType = default(string), string co
         [JsonProperty(PropertyName = "content")]
         public string Content { get; set; }
 
+        /// <summary>
```
Comment (RamjotSingh) [inline CSharp/Microsoft.Bot.Schema.Teams/Generated/MessageActionsPayloadBody.cs]: Do not modify the data model. Text context is something you want only in SDK I am guessing, it should not be in the DTO.

**Comment by RamjotSingh** [comment]
All files in Generated folder are autogenerated from json file. No file in this folder should be touched unless the server<->client models have changed.

File: CSharp/Samples/Microsoft.Bot.Builder.Abstractions.Teams/TeamsActivityProcessor.cs (hunk 4 above)
Comment (jinxiwu) [reply CSharp/Samples/Microsoft.Bot.Builder.Abstractions.Teams/TeamsActivityProcessor.cs]: fixed

File: CSharp/Microsoft.Bot.Schema.Teams/Generated/MessageActionsPayloadBody.cs (hunk 5 above)
Comment (jinxiwu) [reply CSharp/Microsoft.Bot.Schema.Teams/Generated/MessageActionsPayloadBody.cs]: fixed

File: CSharp/Microsoft.Bot.Schema.Teams/MessageActionsPayloadBodyExtensions.cs [hunk 6]
```diff
@@ -0,0 +1,54 @@
...
+        /// <param name="body">The MessageActionsPayloadBody.</param>
+        /// <returns>Plain text content.</returns>
+        public static string GetPlainTextContent(this MessageActionsPayloadBody body)
+        {
+            var doc = new HtmlDocument();
+            doc.LoadHtml(body.content);
+            return StripHtmlTagsHelper(doc.DocumentNode, TextRestrictedHtmlTags);
+        }
+
+        private static readonly HashSet<string> TextRestrictedHtmlTags = new HashSet<string> { "at", "attachment" };
```
Comment (RamjotSingh) [inline CSharp/Microsoft.Bot.Schema.Teams/MessageActionsPayloadBodyExtensions.cs]: This should be on top of the class before methods

File: CSharp/Microsoft.Bot.Schema.Teams/MessageActionsPayloadBodyExtensions.cs [hunk 7]
```diff
@@ -0,0 +1,54 @@
...
+
+        private static string StripHtmlTagsHelper(HtmlNode node, ISet<string> tags)
```
Comment (RamjotSingh) [inline CSharp/Microsoft.Bot.Schema.Teams/MessageActionsPayloadBodyExtensions.cs]: StripHtmlTags

Comment (jinxiwu) [reply CSharp/Microsoft.Bot.Schema.Teams/MessageActionsPayloadBodyExtensions.cs]: done

File: CSharp/Microsoft.Bot.Schema.Teams/MessageActionsPayloadBodyExtensions.cs (hunk 6 above)
Comment (jinxiwu) [reply CSharp/Microsoft.Bot.Schema.Teams/MessageActionsPayloadBodyExtensions.cs]: moved

File: CSharp/Microsoft.Bot.Schema.Teams/MessageActionsPayloadBodyExtensions.cs [hunk 8]
```diff
@@ -0,0 +1,53 @@
...
+        /// <summary>
+        /// Strip HTML tags from MessageActionsPayloadBody content.
+        /// </summary>
+        /// <param name="body">The MessageActionsPayloadBody.</param>
+        /// <returns>Plain text content.</returns>
+        public static string GetPlainTextContent(this MessageActionsPayloadBody body)
+        {
+            var doc = new HtmlDocument();
+            doc.LoadHtml(body.content);
+            var TextRestrictedHtmlTags = new HashSet<string> { "at", "attachment" };
```
[the remark this answers is not in the document]
Comment (jinxiwu) [reply CSharp/Microsoft.Bot.Schema.Teams/MessageActionsPayloadBodyExtensions.cs]: fixed

**Comment by RamjotSingh** [comment]
LGTM

**Pull request merged**